config/README.md

```diff
@@ -5,6 +5,28 @@ iceoryx2 introduces global configuration settings. It enables the concurrent
 execution of multiple iceoryx2 setups on the same machine or within a single
 process by employing distinct configurations.
 
+When **iceoryx2** is started without an explicitly loaded configuration,
+such as:
+
+```rust
+use iceoryx2::config::Config;
+use iceoryx2_bb_system_types::file_path::FilePath;
+use iceoryx2_bb_container::semantic_string::SemanticString;
+
+Config::setup_global_config_from_file(
+    &FilePath::new(b"my/custom/config/file.toml")?)?;
+```
+
+it will automatically search for a configuration file in the following
+locations, in order:
+
+1. `$PWD/config/iceoryx2.toml`
+2. `$HOME/.config/iceoryx2/iceoryx2.toml`
```

Comment:
Btw, is the `iceoryx2` folder part of the config?

If yes, I would make that independent of the config, so the user could set the `domain` or whatever ... hmm, maybe an `uri` would make sense, i.e. `io.iceoryx2`. User could then use their own reverse domain name notation when they want to isolate themself from the general communication. `/tmp/io.iceoryx2` could then hold a list of all used `uri`, which would help the tooling to discover those islands.

Reply:
Currently, the `iceoryx2` is a static fixed part of the config file lookup location. It cannot be part of the config since we do not know the config until we have read it and for this we need to know the full config file path.

Nevertheless, I like the underlying idea and would like to pursue it further. Maybe we could somehow define a domain that is then reflected in the file name or a subdirectory, but `/etc/iceoryx2/` or `$HOME/.config/iceoryx2` is static and unchangeable but the structure inside this folder is open for debate.

Reply:
I created issue #618 to keep track of this idea.